multi-agent/lib/mas/session/repository/repository.py

```
@@ -37,17 +39,28 @@ def list_runs(self, identity: Identity) -> List[str]:
+    ) -> List[Mapping[str, Any]]:
+        """Return paginated session documents sorted by most recent activity, with newest sessions first."""
+        ...
+
     @abstractmethod
     def delete(self, run_id: str) -> bool:
         """Delete a session by run_id. Returns True if deleted, False if not found."""
         ...
 
     @abstractmethod
-    def count(self, identity: Identity, filter: Dict[str, Any]) -> int:
+    def count(self, identity: Identity, filter: Optional[SessionListFilter] = None) -> int:
```

Ruff (0.16.1)

[error] Function argument `filter` is shadowing a Python builtin

(A002)